src/main/java/net/minecraftforge/server/permission/nodes/PermissionNode.java

```diff
@@ -0,0 +1,180 @@
+     *                        They <strong>must</strong> be passed into the constructor if you want to use them.
+     */
+    private PermissionNode(String nodeName, PermissionType<T> type, PermissionResolver<T> defaultResolver, PermissionDynamicContextKey... dynamics)
+    {
+        this.nodeName = nodeName;
+        this.type = type;
+        this.dynamics = dynamics;
+        this.defaultResolver = defaultResolver;
+    }
+
+    /**
+     * Allows you to set a human-readable name & description for your Permission.
+     *
+     * <p>Note: Even though not used by Default, PermissionHandlers may display this information in game,
+     * or provide it to the user by other means.</br>
+     * You may use {@link net.minecraft.network.chat.TranslatableComponent}, but you'll need 2 language files.
+     * One inside the data directory for the server and one inside assets for the client.</p>
+     *
+     * @param readableName an easier to read name for the PermissionNode,
+     *                     when using TranslatableComponent, key should be of format "permission.name.<nodename>"
+     * @param description  description for the PermissionNode
+     *                     when using TranslatableComponent, key should be of format "permission.desc.<nodename>"
+     * @return itself with the new information set.
+     */
+    public PermissionNode setInformation(@Nonnull Component readableName, @Nonnull Component description)
+    {
+        Preconditions.checkNotNull(readableName, "Readable name for PermissionNodes must not be null %s", this.nodeName);
+        Preconditions.checkNotNull(description, "Description for PermissionNodes must not be null %s", this.nodeName);
+
+        this.readableName = readableName;
+        this.description = description;
+
+        return this;
+    }
+
+    public String getNodeName()
+    {
+        return nodeName;
+    }
+
+    public PermissionType<T> getType()
+    {
+        return type;
+    }
+
+    public PermissionDynamicContextKey<?>[] getDynamics()
+    {
+        return dynamics;
+    }
+
+    public PermissionResolver<T> getDefaultResolver()
+    {
+        return defaultResolver;
+    }
+
+    public Component getReadableName()
+    {
+        return readableName;
+    }
+
+    public Component getDescription()
+    {
+        return description;
+    }
+
+    /**
+     * Utility Interface used for resolving the default value of PermissionNodes
+     *
+     * @param <T> generic value of the PermissionType of a PermissionNode
+     */
+    @FunctionalInterface
+    public interface PermissionResolver<T>
+    {
+        T resolve(ServerPlayer player, PermissionDynamicContext... context);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+        if (!(o instanceof PermissionNode)) return false;
+        PermissionNode<?> that = (PermissionNode<?>) o;
```

Comment:
Pattern matching `instanceof` can be used here:
```suggestion
        if (!(o instanceof PermissionNode that)) return false;
```